pkg/controller/rhmiconfig/rhmiconfig_controller.go

```diff
@@ -113,6 +114,28 @@ func (r *ReconcileRHMIConfig) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	//	Checks if there is an upgrade available, if there is upgrade available upgrades the status of RHMIConfig
+	if rhmiConfig.Status.UpgradeAvailable != nil {
+		if reschedule, ok := rhmiConfig.Annotations[integreatlyv1alpha1.RecalculateScheduleAnnotation]; ok && reschedule == "true" {
+			err = helpers.UpdateStatus(context.TODO(), r.client, rhmiConfig)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			rhmiConfig.Annotations[integreatlyv1alpha1.RecalculateScheduleAnnotation] = ""
+			err = r.client.Update(context.TODO(), rhmiConfig)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
```

Comment:
```suggestion		
			err = helpers.UpdateStatus(context.TODO(), r.client, rhmiConfig)
			if err != nil {
				return reconcile.Result{}, err
			}

			rhmiConfig.Annotations[integreatlyv1alpha1.RecalculateScheduleAnnotation] = ""
			err = r.client.Update(context.TODO(), rhmiConfig)
			if err != nil {
				return reconcile.Result{}, err
			}		
```

I realized that this condition can go. There's no need for that annotation anymore as the schedule will only be recalculated when the UpgradeAvailable field is modified

Reply:
:+1: